enterprise/backend/src/metabase_enterprise/metabot_v3/reactions.clj

```diff
@@ -90,8 +89,12 @@
    and return the result to the user."
   #{:metabot.reaction/redirect})
 
+(defn- terminating-reaction?
+  "Checks if a reaction is a terminating reaction"
+  [reaction]
+  (contains? terminating-reaction-types (:type reaction)))
+
 (defn has-terminating-reaction?
   "Checks if the envelope reactions vector contains a terminating reaction"
   [reactions]
-  (let [reaction-types (->> reactions (map :type) set)]
-    (seq (set/intersection terminating-reaction-types reaction-types))))
+  (some terminating-reaction? reactions))
```

Comment:
I would have made `terminating-reaction?` public and inline `has-terminating-reaction?`.